File: student_frontend/lib/model/booking/availability_period.dart
class AvailabilityPeriod {
  final int id;
  final int numAvailable;
  final String fromTime;
  final String toTime;
  final bool overrideDefaultPrice;
  final double basePrice;
  final double pricePerHalfHour;

  AvailabilityPeriod({
    required this.id,
    required this.numAvailable,
    required this.fromTime,
    required this.toTime,
    required this.overrideDefaultPrice,
    required this.basePrice,
    required this.pricePerHalfHour,
  });

  factory AvailabilityPeriod.fromJson(Map<String, dynamic> json) {
    return AvailabilityPeriod(
      id: json['id'],
      numAvailable: json['numAvailable'],
      fromTime: json['fromTime'],
      toTime: json['toTime'],
      overrideDefaultPrice: json['overrideDefaultPrice'],
      basePrice: json['basePrice'],
      pricePerHalfHour: json['pricePerHalfHour'],
    );
  }
}

File: backend/Backend/src/main/java/com/sgw/backend/entity_booking/AvailabilityPeriod.java
package com.sgw.backend.entity_booking;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalTime;

@Entity
@Data
public class AvailabilityPeriod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private int numAvailable;

    @Column(nullable = false)
    @Temporal(TemporalType.TIME)
    private LocalTime fromTime;
    @Column(nullable = false)
    @Temporal(TemporalType.TIME)
    private LocalTime toTime;

    // if they want to override the TableType price. small feature tho, can remove if it doesn't work out
    @Column(nullable = false)
    private boolean overrideDefaultPrice = false;

    @Column(nullable = false)
    private BigDecimal basePrice = BigDecimal.ZERO;
    @Column(nullable = false)
    private BigDecimal pricePerHalfHour;

}